src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -878,11 +866,55 @@ public List<E> reversed() {
             return new StableReverseOrderListView<>(this);
         }
 
+        @Override
+        public List<E> subList(int fromIndex, int toIndex) {
+            int size = size();
+            subListRangeCheck(fromIndex, toIndex, size);
+            return StableSubList.fromList(this, fromIndex, toIndex);
+        }
+
         @Override
         public String toString() {
             return StableUtil.renderElements(this, "StableList", delegates);
         }
 
+        private static final class StableSubList<E> extends SubList<E> {
+
+            private StableSubList(AbstractImmutableList<E> root, int offset, int size) {
+                super(root, offset, size);
+            }
+
+            @Override
+            public List<E> reversed() {
+                return new StableReverseOrderListView<>(this);
+            }
+
+            @Override
+            public List<E> subList(int fromIndex, int toIndex) {
+                int size = size();
+                subListRangeCheck(fromIndex, toIndex, size);
+                return StableSubList.fromSubList(this, fromIndex, toIndex);
+            }
+
+            @Override
+            public String toString() {
+                final StableList<E> deepRoot = deepRoot(root);
+                final StableValueImpl<E>[] delegates = deepRoot.delegates;
+                // Todo: Provide a copy free solution
+                final StableValueImpl<E>[] reversed = Arrays.copyOfRange(delegates, this.offset, this.size - offset);
+                return StableUtil.renderElements(this, "StableCollection", reversed);
```

Comment:
```suggestion
                return StableUtil.renderElements(this, "StableCollection", 
                                                 Arrays.copyOfRange(deepRoot(root).delegates, this.offset, this.size - offset));
```
The three local variables `deepRoot/delegates/reversed` are only used once. Can they be simplified like this?